pulsar-common/src/main/java/org/apache/pulsar/common/util/collections/ConcurrentLongLongPairHashMap.java

```diff
@@ -322,15 +324,15 @@ private static final class Section extends StampedLock {
         LongPair get(long key1, long key2, int keyHash) {
             long stamp = tryOptimisticRead();
             boolean acquiredLock = false;
-            int bucket = signSafeMod(keyHash, capacity);
+            int bucketIndex = signSafeMod(keyHash, capacity) * ITEM_SIZE;
```

Comment:
The problem with the old code is that it is cryptic. In the original code, `signSafeMod` function's name is misleading. In the cases where the item takes 2 or 4 elements, there's a bitwise shift to left to multiply by 2 (`<< 1`) or 4 (`<< 2`). I doubt that using bitwise operations to optimize the speed of execution is worthwhile in Java code. It should be left to the compiler to do such optimizations.

Reply:
Change method name `signSafeMod` to `hash` may be better, right? 
And i wonder whether the compiler will do such optimizations, could you verify this? thanks.
IMO, these util class adopt some kind of ideas from common util class like `HashMap`, which use `&` to avoid `%` too.

Reply:
The modern Java compilers are smart enough to recognize when multiplication is equivalent to a bit shift and can optimize the multiplication operation to use a bit shift instead. It's better to focus on writing clear, readable code.